### PR TITLE
Add cli and tests for metastore summary API

### PR DESCRIPTION
**Pull request by vikrantpuppala**
**PR Checklist**

- [ ] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Add tests for metastore summary API

### CODE REVIEW

File: server/src/test/java/io/unitycatalog/server/base/metastore/BaseMetastoreSummaryTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  @Override
+  public void setUp() {
+    super.setUp();
+    metastoreOperations = createMetastoreOperations(serverConfig);
+  }
+
+  @Test
+  public void testMetastoreSummary() throws ApiException {
+    GetMetastoreSummaryResponse metastoreSummary = metastoreOperations.getMetastoreSummary();
+    assertThat(metastoreSummary.getMetastoreId()).isNotNull();
```
Comment (sumeet-db) [inline server/src/test/java/io/unitycatalog/server/base/metastore/BaseMetastoreSummaryTest.java:27]: Can we call it again and verify the response remains stable?

File: server/src/test/java/io/unitycatalog/server/base/metastore/BaseMetastoreSummaryTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  @Test
+  public void testMetastoreSummary() throws ApiException {
+    GetMetastoreSummaryResponse metastoreSummary = metastoreOperations.getMetastoreSummary();
+    String metastoreId1 = metastoreSummary.getMetastoreId();
+    assertThat(metastoreId1).isNotNull();
+    assertThat(metastoreSummary.getName()).isNotNull();
+    // Verify that it is stable
+    String metastoreId2 = metastoreSummary.getMetastoreId();
+    assertThat(metastoreId2).isNotNull();
+    assertThat(metastoreId1).isEqualTo(metastoreId2);
```
Comment (sumeet-db) [inline server/src/test/java/io/unitycatalog/server/base/metastore/BaseMetastoreSummaryTest.java:33]: Check `createdAt` and `updatedAt` are non-null as well?

**Review by sumeet-db — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -26,6 +26,7 @@ etc/conf/*.der
 connectors/spark/etc/conf/certs.json
 connectors/spark/etc/conf/*.txt
 connectors/spark/etc/conf/*.der
+.java-version
```
Comment (sumeet-db) [inline .gitignore:29]: (Assuming this will be moved to a separate PR.)

File: examples/cli/src/main/java/io/unitycatalog/cli/MetastoreCli.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package io.unitycatalog.cli;
```
Comment (tdas) [inline examples/cli/src/main/java/io/unitycatalog/cli/MetastoreCli.java:1]: can you make a separate PR for the doc update? lets put [0.3] in the PR title so that we can find and merge it when we are closer to 0.3 release

File: examples/cli/src/main/java/io/unitycatalog/cli/MetastoreCli.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      case CliUtils.GET:
+        output = getMetastoreSummary(metastoresApi);
+        break;
+      default:
+        CliUtils.printEntityHelp(CliUtils.METASTORE);
+    }
+    postProcessAndPrintOutput(cmd, output, subCommand);
+  }
+
+  private static String getMetastoreSummary(MetastoresApi metastoresApi)
```
[the remark this answers is not in the document]
Comment (vikrantpuppala) [reply examples/cli/src/main/java/io/unitycatalog/cli/MetastoreCli.java:33]: ah, was just following convention with other Cli classes like [this](https://github.com/unitycatalog/unitycatalog/blob/main/examples/cli/src/main/java/io/unitycatalog/cli/CatalogCli.java#L81-L85).. do you prefer it to be inline?

File: examples/cli/src/main/java/io/unitycatalog/cli/MetastoreCli.java (hunk 4 above)
Comment (vikrantpuppala) [reply examples/cli/src/main/java/io/unitycatalog/cli/MetastoreCli.java:1]: will do!

File: examples/cli/src/main/java/io/unitycatalog/cli/MetastoreCli.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (tdas) [reply examples/cli/src/main/java/io/unitycatalog/cli/MetastoreCli.java:33]: convention seems like an overkill here.

[the remark this answers is not in the document]
Comment (vikrantpuppala) [reply examples/cli/src/main/java/io/unitycatalog/cli/MetastoreCli.java]: fixed!

**Review by tdas — APPROVED** [review]

**Comment by vikrantpuppala** [comment]
@tdas the automerge on this one got disabled
